bitnami/common/templates/_secrets.tpl

```diff
@@ -100,7 +100,7 @@ The order in which this function returns a secret password:
 {{- if $secretData }}
   {{- if hasKey $secretData .key }}
     {{- $password = index $secretData .key | quote }}
-  {{- else if $failOnNew }}
+  {{- else if eq $failOnNew "true" }}
```

Comment:
Before discarding using booleans, could you try this?

- Remove use or `default` to set `$failOnNew`:

```diff
-{{- $failOnNew := default "true" .failOnNew }}
```

- Use `if not` here:

```suggestion
  {{- else if not .failOnNew }}
```

Then, keep using a boolean on Kafka chart.

Reply:
Wouldn't that mean the `failOnNew` can not be set to false?

Also in my test I was not able to use `failOnNew` without Helm asking to provide the current password via additional `--set` parameter.

Could you verify if it is working for you?